rfcs/kubernetes-oci/README.md

```diff
@@ -0,0 +1,428 @@
+# RFC-xxxx Flux OCI support for Kubernetes manifests
+
+**Status:** provisional
+
+**Creation date:** 2022-03-31
+
+**Last update:** 2022-06-22
+
+## Summary
+
+Flux should be able to distribute and reconcile Kubernetes configuration packaged as OCI artifacts.
+
+On the client-side, the Flux CLI should offer a command for packaging Kubernetes configs into
+an OCI artifact and pushing the artifact to a container registry using the Docker config file
+and the Docker credential helpers for authentication.
+
+On the server-side, the Flux source-controller should offer a dedicated API Kind for defining
+how OCI artifacts are pulled from container registries and how the artifact's authenticity can be verified.
+Flux should be able to work with any type of artifact even if it's not created with the Flux CLI.
+
+## Motivation
+
+Given that OCI registries are evolving into a generic artifact storage solution,
+we should extend Flux to allow fetching Kubernetes manifests and related configs
+from container registries similar to how Flux works with Git and Bucket storage.
+
+With OCI support, Flux users can automate artifact updates to Git in the same way
+they do today for container images.
```

Comment:
> With OCI support, Flux users can automate artifact updates to Git in the same way
> they do today for container images.

Not sure I understand what workflow we are referring here. Is this referring to updating Git as in what IAC does. Or is this about using OCI to trigger source-controller updates, pretty much as SC does for GitRepository objects?

Reply:
It refers to IAC, where Flux patches the `spec.ref.tag` in the same way it does today for `images[].newTag`.